ivy/container/base.py

```diff
@@ -149,6 +149,7 @@ def __init__(
         )
         self._config = dict()
         self.inplace_update(dict_in, **self._config_in)
+        self._is_variable = ivy.is_variable(dict_in)
```

Comment:
`dict_in` will never be a variable, so this check doesn't make sense. If we want `self._is_variable` to be an `ivy.Container` with bools at the leaves, then we should cast `dict_in` to an `ivy.Container` before calling this function, otherwise `ivy.is_variable` will not detect any `ivy.Container` instances in the input, and therefore will not be called in a nestable manner.

Reply:
Sorry I meant to write `dict_in[key]`. 

For example, say we create `x = ivy.Container({"w": torch.tensor([3.0],requires_grad=True)})`, so `is_variable(x)` should evaluate to `True` because the leaf is a torch tensor with `requires_grad=True`. Therefore, we can have `self._is_variable = ivy.is_variable(dict_in[key])` to extract the values out of the `dict_in`.

But I'm wondering what if we have a container such as `Container({"w1": torch.tensor([3.0],requires_grad=True)}, {"w2":np.array([2.0])})`, where we have mixed structures ,and numpy does not support auto differentiation.

Or we just do not consider a container to be a `Variable` at all?

Reply:
there is no `key` variable passed into the constructor, and so adding `dict_in[key]` wouldn't work. If we want `self._is_variable` to be a container (which makes more sense), then we can just change this to `self._is_variable = ivy.is_variable(self)` at the end of the constructor.

Reply:
sorry for being unclear. The "key" isn't a key literal, I meant the keys for the `dict_in`, like `w1` and `w2` as I gave in the second example.
If we use `self._is_variable = ivy.is_variable(self)`, is there any method to deal with internally conflicting inputs such as `cont = ivy.Container({"w1": torch.tensor([3.0],requires_grad=True)}, {"w2":np.array([2.0])})`, where `is_variable(cont["w1"])=True` but `is_variable(cont["w2"])=False`. Will we perform broadcasts in this case?
I guess it's more of a design pattern to decide if a container can be treated as a variable as a whole that can support gradient computation, or if we would rather consider leaves separately.
Please correct me if I'm wrong

Reply:
we should definitely consider leaves separately. Also, Ivy does not support mixed backend frameworks, and so `cont = ivy.Container({"w1": torch.tensor([3.0],requires_grad=True)}, {"w2":np.array([2.0])})` is something that we should not expect to deal with, but we should indeed support variables and non-variables at the leaves, such as `cont = ivy.Container({"w1": torch.tensor([3.0],requires_grad=True)}, {"w2":torch.tensor([2.0])})`